src/lib.rs

```diff
@@ -198,30 +192,36 @@ impl Serializable<16> for BalanceInfo {
         Ok(Self { value, spendable })
     }
 
+    #[allow(unused_must_use)]
     fn to_bytes(&self) -> [u8; Self::SIZE] {
         let mut buf = [0u8; Self::SIZE];
+        let mut writer = &mut buf[..];
 
-        buf[0..8].copy_from_slice(&self.value.to_bytes());
-        buf[8..16].copy_from_slice(&self.spendable.to_bytes());
+        writer.write(&self.value.to_bytes());
+        writer.write(&self.spendable.to_bytes());
 
         buf
     }
 }
 
 /// The stake of a particular key.
-#[derive(Debug, Hash, Clone, Copy, PartialEq, Eq)]
+#[derive(Debug, Default, Hash, Clone, Copy, PartialEq, Eq)]
 pub struct StakeInfo {
-    /// The amount staked.
-    pub value: u64,
-    /// Block by which the stake is valid.
-    pub eligibility: u64,
-    /// Block by which stake was created.
-    pub created_at: u64,
+    /// The value and eligibility of the stake, in that order.
+    pub amount: Option<(u64, u64)>,
+    /// The reward available for withdrawal.
+    pub reward: u64,
+    /// Signature counter.
+    pub counter: u64,
```

Comment:
Unrelated to this PR, but shouldn't this contain also the stake key? As a mandatory part of any stake.